### PR TITLE
feat: configre the split methods

### PR DESCRIPTION
**Pull request by chrfritsch**

### CODE REVIEW

File: src/Form/ParagraphsPasteForm.php [hunk 1]
```diff
@@ -131,13 +131,23 @@ public static function pasteSubmit(array $form, FormStateInterface $form_state)
...
+    }
+    if (!empty($settings['split_method']['oEmbed'])) {
+      $reg_ex[] = "https?://[^\s/$.?#].[^\s]*";
+    }
+    if (!empty($settings['split_method']['regex'])) {
+      $reg_ex[] = $settings['split_method']['regex'];
+    }
+
+    // Split by RegEx.
+    $data = preg_split('~(' . implode('|', $reg_ex) . ')~', $pasted_data, -1, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
```
Comment (ol0lll) [inline src/Form/ParagraphsPasteForm.php:149]: We need to bail out when $reg_ex is empty, the php process is not coming back (504 gateway timeout) when no split method is set.

File: src/Form/ParagraphsPasteForm.php [hunk 2]
```diff
@@ -248,11 +258,37 @@ public static function getThirdPartyForm(WidgetInterface $plugin, $field_name) {
...
+    $elements['split_method'] = [
+      '#type' => 'fieldset',
+      '#title' => t('Split methods'),
+      '#description' => t('Define when new paragraphs should be created.'),
+      '#states' => ['visible' => [":input[name=\"fields[$field_name][settings_edit_form][third_party_settings][paragraphs_paste][enabled]\"]" => ['checked' => TRUE]]],
+    ];
+
+    $elements['split_method']['paragraph'] = [
+      '#type' => 'checkbox',
+      '#title' => 'By text paragraph',
```
Comment (ol0lll) [inline src/Form/ParagraphsPasteForm.php]: Actually its split by double newline and please activate as default.

File: src/Form/ParagraphsPasteForm.php [hunk 3]
```diff
@@ -131,13 +131,23 @@ public static function pasteSubmit(array $form, FormStateInterface $form_state)
...
+    }
+    if (!empty($settings['split_method']['regex'])) {
+      $reg_ex[] = $settings['split_method']['regex'];
+    }
+
+    // Split by RegEx.
+    $data = preg_split('~(' . implode('|', $reg_ex) . ')~', $pasted_data, -1, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
 
-    $items = self::traverseData($data, $property_path_mapping);
+    $items = self::traverseData($data, $settings['mapping']);
```
Comment (ol0lll) [inline src/Form/ParagraphsPasteForm.php:151]: I think the plugin needs information about the split method. 

Say, the stuff was split on "h2:" this is lost on preg_split, mylife plugin would not be able to put the line into  heading field.

**Review by ol0lll — CHANGES_REQUESTED** [review]

File: src/Form/ParagraphsPasteForm.php (hunk 3 above)
Comment (chrfritsch) [reply src/Form/ParagraphsPasteForm.php:151]: Shouldn't that be handled in the PR where we try to implement that?

Comment (chrfritsch) [reply src/Form/ParagraphsPasteForm.php:151]: Shouldn't that be handled in the PR where we try to implement that?

Comment (ol0lll) [reply src/Form/ParagraphsPasteForm.php:151]: I just thought you might want to keep the naming rather than changing it back in a few PRs ;-)

**Review by ol0lll — APPROVED** [review]
Lgtm

**Pull request merged**